pkg/pmem-state/pmme-state.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2019  Intel Corporation.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+package pmemstate
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/pkg/errors"
+	"k8s.io/klog/glog"
+)
+
+// GetAllFunc callback function used for StateManager.GetAll().
+// This function is called with ID, for each entry found in the state.
+type GetAllFunc func(id string) bool
+
+// StateManager manages the driver persistent state, i.e, volumes information
+type StateManager interface {
+	// Create creates an entry in the state with given id and data
+	Create(id string, data interface{}) error
+	// Delete deletes an entry found with the id from the state
+	Delete(id string) error
+	// Get retrives the entry data into location pointed by dataPtr.
+	Get(id string, dataPtr interface{}) error
+	// GetAll retrieves all entries found in the state, foreach functions is
+	// called with id for every entry found in the state, and entry data is filled in dataPtr.
+	// the caller has to copy the data if needed.
+	GetAll(dataPtr interface{}, foreach GetAllFunc) error
+}
+
+// fileState Persists the state information into a file.
+// This is is supposed to use by Nodes to persists the state.
+type fileState struct {
+	location string
+}
+
+var _ StateManager = &fileState{}
+
+// NewFileState instantiates the file state manager with given directory
+// location. It ensures the provided directory exists.
+// Returns error, if fails to create the direcotry incase of not pre-existing.
+func NewFileState(loc string) (StateManager, error) {
```

Comment:
If `loc` must be a directory, why not call it `dir`? Same for `ensureLocation`. Or perhaps `directory`, depending whether you prefer abbreviations or not.

Reply:
Done